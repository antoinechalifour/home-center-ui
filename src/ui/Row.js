import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Row = ({ children, ...props }) => {
  const kids = React.Children.map(children, child => {
    if (child.props.flex) {
      const flex = typeof child.props.flex === 'number' ? child.props.flex : 1
      const style = child.props.style ? { ...child.props.style } : {}

      style.flex = flex

      return React.cloneElement(child, { style })
    } else {
      return child
    }
  })

  return <Wrapper {...props}>{kids}</Wrapper>
}

Row.propTypes = {
  align: PropTypes.string.isRequired,
  justify: PropTypes.string.isRequired
}

Row.defaultProps = {
  align: 'flex-start',
  justify: 'flex-start'
}

const Wrapper = styled.div`
  display: flex;
  align-items: ${({ align }) => align};
  justify-content: ${({ justify }) => justify};
`

export default Row
