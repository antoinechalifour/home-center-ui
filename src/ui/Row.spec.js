import React from 'react'
import { mount } from 'enzyme'
import Row from './Row'

const tests = [
  { align: 'flex-start', justify: 'center' },
  { align: 'flex-end', justify: 'flex-start' },
  { align: 'flex-start', justify: 'space-around' }
]

describe('ui/Row', () => {
  tests.forEach(props =>
    it(`Should render`, () => {
      const wrapper = mount(
        <Row {...props}>
          <div>Hey</div>
          <div>Oh</div>
          <div>Lets go!</div>
        </Row>
      )

      expect(wrapper).toMatchSnapshot()
    })
  )
})
