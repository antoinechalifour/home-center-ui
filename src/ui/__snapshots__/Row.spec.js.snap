// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Row Should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<Flex(row)
  align="flex-start"
  justify="center"
>
  <styled.div
    align="flex-start"
    direction="row"
    justify="center"
  >
    <div
      className="c0"
      direction="row"
    >
      <div
        key=".0"
      >
        Hey
      </div>
      <div
        key=".1"
      >
        Oh
      </div>
      <div
        key=".2"
      >
        Lets go!
      </div>
    </div>
  </styled.div>
</Flex(row)>
`;

exports[`ui/Row Should render 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<Flex(row)
  align="flex-end"
  justify="flex-start"
>
  <styled.div
    align="flex-end"
    direction="row"
    justify="flex-start"
  >
    <div
      className="c0"
      direction="row"
    >
      <div
        key=".0"
      >
        Hey
      </div>
      <div
        key=".1"
      >
        Oh
      </div>
      <div
        key=".2"
      >
        Lets go!
      </div>
    </div>
  </styled.div>
</Flex(row)>
`;

exports[`ui/Row Should render 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<Flex(row)
  align="flex-start"
  justify="space-around"
>
  <styled.div
    align="flex-start"
    direction="row"
    justify="space-around"
  >
    <div
      className="c0"
      direction="row"
    >
      <div
        key=".0"
      >
        Hey
      </div>
      <div
        key=".1"
      >
        Oh
      </div>
      <div
        key=".2"
      >
        Lets go!
      </div>
    </div>
  </styled.div>
</Flex(row)>
`;
